pentaho/impl/src/test/java/pt/webdetails/cdf/dd/testUtils/StringHeaderHttpServletResponseForTests.java

```diff
@@ -0,0 +1,196 @@
+/*!
+ * Copyright 2019 Webdetails, a Hitachi Vantara company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to  http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+package pt.webdetails.cdf.dd.testUtils;
+
+import pt.webdetails.cpf.web.DelegatingServletOutputStream;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintWriter;
+import java.util.Collection;
+import java.util.Hashtable;
+import java.util.Locale;
+
+/**
+ * This class is used for unit test only and is used to test the buffer, headers and status. The headers are strings
+ * only. This class was needed to retrieve header values for testing.
+ */
+public class StringHeaderHttpServletResponseForTests implements HttpServletResponse {
```

Comment:
Have you looked if `org.springframework.mock.web.MockHttpServletResponse` would be enough for your needs?

Reply:
I only saw the MockHttpServletResponse object that was made in CPF project, which has issues with conflicting javax servlet api.  I can look into the spring framework one, but I believe that would be an added dependency, as cde does not target that dependency as of now.  I will look into the springframework mock.

Reply:
Spring Framework class is much nicer to work with. I swapped the code out and removed the test util class.

Reply:
Spring Framework class is much nicer to work with. I swapped the code out and removed the test util class.